javascript/javascript3/week3/ScreenShots/script.js

```diff
@@ -0,0 +1,254 @@
+// initializing variables
+// Decided not to hide crudcrudKEY anywhere - it is valid for only 24 hours
+const screenshotKEY = "af35d9bda9msh1d5d601753355fcp1d5ca1jsnc37e78a50099";
+const crudcrudKEY = "3454dc8fc29e4c269600bb3028ff235f";
+const crudcrudURL = "https://crudcrud.com/api";
+const baseSCR_URL = "https://website-screenshot6.p.rapidapi.com/screenshot";
+let currentUser = {
+  email: "john@smith",
+  hash: -946852072
+};
+
+/**
+ * Deletes the screenshot record from crudcrud.
+ * @param {hex} id - unique number to identify user. the format provided by crudcrud.com
+ */
+async function deleteScreenShot(id) {
+  const response = await fetch(
+    `${crudcrudURL}/${crudcrudKEY}/screenshots/${id}`,
+    {
+      method: "DELETE"
+    }
+  );
+  console.dir(response.status);
+  document.getElementById(id).remove();
+}
+
+/**
+ * Gets the target url, then calls website-screenshot6.p.rapidapi.com to get the link with the image.
+ * Posts the screenshot and metadata to crudcrud by calling postScreenshot.
+ * Then calls renderScreenshot to show it on the page.
+ * @param {object} event - form submit.
+ */
+async function getScreenShot(event) {
+  event.preventDefault();
```

Comment:
In my understanding, there is no event that should be stopped from propagating. Maybe there is something I am not taking into consideration.
event.preventDefault() is usually associated with form submissions, and you will see it a lot in React when handling form events.

Reply:
I indeed call this function upon a form submit (I structured the fields and buttons as the fields of the form), so I need this in order to avoid autorefresh on a page.